### ISSUE TITLE
Paste from Google Docs inserts non-breaking space between paragraphs

### ISSUE THREAD

**Issue by Lithl** [knighty/fimfiction-issues#140]
A Google doc with a space between paragraphs, copied and pasted into the new FimFiction editor, adds a non-breaking space where an empty line should be. So, for example, a doc with:

    Lorem ipsum

    dolor sit amet

When pasted into FimFiction, comes out as:

    Lorem ipsum
    \u00a0
    dolor sit amet

The paragraph counter at the bottom of the text area then says that it's one paragraph. For a normal word count of a chapter, this will also trigger the "You have very few paragraphs" warning.

A line with text that isn't a paragraph (eg, a horizontal rule) will count as a paragraph break, so for example my 2,142-word chapter I just pasted in with two horizontal rules is claimed to be 3 paragraphs.

**Comment by knighty** [comment]
Can you link me a gDoc this happens on? Because it doesn't happen on any of mine.

edit: Ok apparently it is not. I think google might have changed their format AGAIN.

**Comment by Lithl** [comment]
Here's the actual doc I just tried it with: https://docs.google.com/document/d/1_FqvQun3S_7jBeCJQUjV6WXFJe2YAPqTeEucGRYflIA/edit?usp=sharing

**Comment by knighty** [comment]
Fortunately it seems like this is only a problem with client side paragraph count. It gets converted fine. I just need to include more whitespace tokens for the client side stuff

**Comment by Lithl** [comment]
Yes, the rendered output doesn't include the extra space between paragraphs. However, even after saving, going to another page, and coming back, the space is still there when editing, when it shouldn't be there at all.

**Comment by knighty** [comment]
Well, the thing is it /is/ in the output from gdocs. Unless I special case gdocs I can't really "fix" that. The solution is just fixing the client side paragraph count.

**Comment by knighty** [comment]
For reference sake https://i.imgur.com/z1W01Gs.png

**Comment by Lithl** [comment]
What about `text.split('\n').map(s => s.replace(/\s+$/, '')).join('\n')` when you're capturing the paste? There's really no need for trailing whitespace, yeah? Depending on how indented paragraphs are handled, you might even be able to strip all whitespace from the start and end of each line, too.

**Comment by knighty** [comment]
I guess I could do a post process pass of some kind on the output. Really I just wish these services would stop giving me horrible html...

**Comment by Lithl** [comment]
I bet its an artifact of contenteditable; I've run in to similar ugly HTML with contenteditable before, too.

Besides, aren't you already doing some kind of processing on the pasted text in order to add in BBCode as needed?
